internal/pkg/util/ociauth/ociauth.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2023, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package ociauth
```

Comment:
I'm not sure there is a good reason not to keep this with other OCI related code in `pkg/remote/credential`?

Reply:
Just a general hesitancy to put things in toplevel `pkg/` without having a good long think about it first...

Reply:
Sorry, now realizing that you meant `*internal*/pkg/remote/credential` - in which case, yes, there's a reason: the stuff in ociauth.go also gets used in internal/pkg/client/ocisif/ocisif.go, so it made sense (to me at least) to have it in a place that's related to both consumers (internal/pkg/client/ocisif/ocisif.go and internal/pkg/remote/credential/login_handler.go) in a symmetrical way.

Reply:
Sorry about the missing `internal`.

Code that's called from two places doesn't always have to be moved out to a third place for caller symmetry. The package `internal/pkg/remote/credential` is specifically about handling/storing OCI and other credentials, so is an appropriate place.

In Go code packages are often quite large, and things are internally split through multiple source files within the package, rather than many smaller packages being created.

I would agree about splitting, however, if the code is buried in the UI layer. I.E. if this code had been buried in the `cmd` or `app` portions then I'd definitely have agreed about moving it... as we want a logical separation of interface/CLI and lower level implementation stuff.